### PR TITLE
Add project support for lxd_container and lxd_profile module

### PR DESCRIPTION
**Pull request by we10710aa**
##### SUMMARY

This PR add [lxd project](https://linuxcontainers.org/lxd/docs/master/projects/) support to 'lxd_container' and 'lxd_profile' module

A new module parameter 'project' is added, if the parameter is set module will perform operations on the selected project.

This is related to:
* #4475 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

* lxd_container
* lxd_profile
##### ADDITIONAL INFORMATION

This change is backward compatible, if project parameter is not set,  query parameter will not be added.

Tested on Ubuntu 20.04 with LXD 4.24 including:
* instance without project
* instance with project
* profile without project
* profile with project

### CODE REVIEW

File: plugins/modules/cloud/lxd/lxd_container.py [hunk 1]
```diff
@@ -21,6 +21,12 @@
           - Name of an instance.
         type: str
         required: true
+    project:
+        description:
+          - 'Project of an instance.
+            See U(https://github.com/lxc/lxd/blob/master/doc/projects.md).'
+        required: false
+        type: str
```
Comment (felixfontein) [inline plugins/modules/cloud/lxd/lxd_container.py:29]:
```suggestion
        type: str
        version_added: 4.8.0
```

File: plugins/modules/cloud/lxd/lxd_profile.py [hunk 2]
```diff
@@ -21,6 +21,12 @@
           - Name of a profile.
         required: true
         type: str
+    project:
+        description:
+         - 'Project of a profile.
+           See U(https://github.com/lxc/lxd/blob/master/doc/projects.md).'
+        type: str
+        required: false
```
Comment (felixfontein) [inline plugins/modules/cloud/lxd/lxd_profile.py:29]:
```suggestion
        required: false
        version_added: 4.8.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

I've added some very first comments below.

File: changelogs/fragments/4479-add-project-support-for-lxd_container-and-lxd_profile.yml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+minor_changes:
+  - lxd_container - adds ``project`` option to allow selecting project for lxd instance (https://github.com/ansible-collections/community.general/pull/4479).
+  - lxd_profile - adds ``project`` option to allow selecting project for lxd profile (https://github.com/ansible-collections/community.general/pull/4479).
```
Comment (felixfontein) [inline changelogs/fragments/4479-add-project-support-for-lxd_container-and-lxd_profile.yml:3]: Isn't the official spelling `LXD` (https://linuxcontainers.org/)?
```suggestion
  - lxd_container - adds ``project`` option to allow selecting project for LXD instance (https://github.com/ansible-collections/community.general/pull/4479).
  - lxd_profile - adds ``project`` option to allow selecting project for LXD profile (https://github.com/ansible-collections/community.general/pull/4479).
```

**Review by felixfontein — COMMENTED** [review]
Besides that, looks good to me.

File: changelogs/fragments/4479-add-project-support-for-lxd_container-and-lxd_profile.yml (hunk 3 above)
Comment (we10710aa) [reply changelogs/fragments/4479-add-project-support-for-lxd_container-and-lxd_profile.yml]: Thanks, fixed the spelling

**Comment by felixfontein** [comment]
Great! I hope that @conloos finds some time to take a look at this as well, since I'm not using these modules :)

**Comment by conloos** [comment]
Hi Felix,

i was at holidays. I take a look this weekend.

Thanks Frank

File: plugins/modules/cloud/lxd/lxd_profile.py [hunk 4]
```diff
@@ -129,6 +136,18 @@
             parent: br0
             type: nic
 
+# An example for creating a profile in project mytestproject
```
Comment (conloos) [inline plugins/modules/cloud/lxd/lxd_profile.py:139]: I think that you have a "project" to that example. :)

**Comment by conloos** [comment]
Hi @all,

I've completed the review and run a few tests. Everything looks very good. But since I only have one server at home with a small setup, I could only check that it didn't break any of my standard setups.

Greeting Frank
PS: @we10710aa -> i leave a comment to your example
PPS: @felixfontein: from my site - "ship it" after review of the example

File: plugins/modules/cloud/lxd/lxd_profile.py [hunk 5]
```diff
@@ -129,6 +136,18 @@
...
+- hosts: localhost
+  connection: local
+  tasks:
+    - name: Create a profile
+      community.general.lxd_profile:
+        name: testprofile
```
Comment (felixfontein) [inline plugins/modules/cloud/lxd/lxd_profile.py:145]: I think this should fix the example according to @conloos's comment:
```suggestion
        name: testprofile
        project: mytestproject
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@we10710aa thanks for your contribution!
@conloos thanks for reviewing!

**Pull request merged**